sos/report/plugins/ceph_rgw.py

```diff
@@ -0,0 +1,100 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+from socket import gethostname
+import os
+
+
+class CephRGW(Plugin, RedHatPlugin, UbuntuPlugin):
+
+    short_desc = 'CEPH rgw'
+
+    plugin_name = 'ceph_rgw'
+    profiles = ('storage', 'virt', 'container')
+    containers = ('ceph-rgw.*',)
+    ceph_hostname = gethostname()
+
+    services = (
+        'ceph-radosgw@*',
+    )
+
+    def check_enabled(self):
+        # first check services running
+        if super(CephRGW, self).check_enabled():
+            return True
+
+        # also check non empty folders in /var/lib/ceph/radosgw
+        rgw_folder = '/var/lib/ceph/radosgw'
+        if os.path.exists(rgw_folder) and os.path.isdir(rgw_folder):
+            inner_files = os.listdir(rgw_folder)
+            for item in inner_files:
+                full_path = rgw_folder + "/" + item
+                if(os.path.isdir(full_path)):
+                    # verify the folder has files
+                    if not os.listdir(full_path):
+                        continue
+                    else:
+                        return True
+        return False
+
+    def setup(self):
+        # all_logs = self.get_option("all_logs")
+
+        self.add_file_tags({
+            '/var/log/ceph/ceph-client.rgw*.log': 'ceph_rgw_log',
+        })
+
+        self.add_copy_spec([
+                "/var/log/ceph/ceph-client.rgw*.log",
+        ])
```

Comment:
Since we have a single spec here, this can be condensed:

```python
self.add_copy_spec('/var/log/ceph/ceph-cleint.rgw*.log', tags='ceph_rgq_log')
```

Reply:
ack thank you